diesel/src/pg/types/json.rs

```diff
@@ -0,0 +1,115 @@
+//! Support for JSON and `jsonb` values under PostgreSQL.
+
+extern crate serde_json;
+
+use std::io::prelude::*;
+use std::error::Error;
+
+use pg::Pg;
+use types::{self, ToSql, IsNull, FromSql};
+
+primitive_impls!(Json -> (serde_json::Value, pg: (114, 199)));
+primitive_impls!(Jsonb -> (serde_json::Value, pg: (3802, 3807)));
```

Comment:
Unfortunately, my only source for the magic numbers was Googling around and looking at what other libraries did, then confirming that it worked. :-/ I'll keep looking.